sklearn/metrics/tests/test_common.py

```diff
@@ -1013,13 +1013,28 @@ def check_sample_weight_invariance(name, metric, y1, y2):
 
 def test_sample_weight_invariance(n_samples=50):
     random_state = check_random_state(0)
+    # regression
+    y_true = random_state.random_sample(size=(n_samples,))
+    y_pred = random_state.random_sample(size=(n_samples,))
+    y_score = random_state.random_sample(size=(n_samples,))
+    for name in ALL_METRICS:
+        if name not in REGRESSION_METRICS:
+            continue
+        if (name in METRICS_WITHOUT_SAMPLE_WEIGHT or
+                name in METRIC_UNDEFINED_BINARY):
```

Comment:
Both conditions or just the latter (metric_undefined_binary)? The former is used because when removed regression tests fail